base_dav/models/dav_collection.py

```diff
@@ -117,17 +159,35 @@ def to_vobject(self, record):
             )
         return result
 
+    @api.model
+    def _from_vobject_datetime(self, item):
+        if isinstance(item.value, (datetime, date)):
+            return item.value.strftime(tools.DEFAULT_SERVER_DATETIME_FORMAT)
+        return None
+
+    @api.model
+    def _from_vobject_date(self, item):
+        if isinstance(item.value, (datetime, date)):
+            return item.value.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)
+        return None
+
+    @api.model
+    def _from_vobject_binary(self, item):
+        return item.value.encode('ascii')
+
+    @api.model
+    def _from_vobject_char_n(self, item):
+        return item.family
+
     @api.multi
     def _to_vobject_datetime(self, record, field_name):
         result = fields.Datetime.from_string(record[field_name])
-        # TODO: this still generates wrong times
-        result.replace(tzinfo=vobject.icalendar.utc)
-        return result
+        return result.replace(tzinfo=tz.UTC)
 
     @api.multi
     def _to_vobject_datetime_rev(self, record, field_name):
         return record[field_name] and record[field_name]\
-            .replace('-', '').replace(' ', 'T').replace(':', '') + 'Z'
+            .replace('-', '').replace(' ', 'T').replace(':', '')
```

Comment:
why this? According to the spec, this is a timestamp: https://tools.ietf.org/html/rfc6350#section-6.7.4

Reply:
This was for playing around and I forgot to revert it